wrapanapi/openstack.py

```diff
@@ -821,15 +822,16 @@ def does_vm_exist(self, name):
     def remove_host_from_cluster(self, hostname):
         raise NotImplementedError('remove_host_from_cluster not implemented')
 
-    def get_first_floating_ip(self):
+    def get_first_floating_ip(self, pool=None):
         try:
-            self.api.floating_ips.create()
-        except os_exceptions.NotFound:
-            self.logger.error('No more Floating IPs available, will attempt to grab a free one')
+            self.api.floating_ips.create(pool=pool)
```

Comment:
Paraphrasing from Rocketchat:

Since this is deprecated anyway, if you're up for it Leo I'd like to see a a new `neutron_api` property, and it used for the `self.neutron_api.floating_ips.create()` call needed here.

This may require some changes in other functions, for everything having to do with networking moved to `neutron_client`, and be a larger scope than what you would like to take on.

Thoughts?

The neutronclient documentation is really thin here:
https://docs.openstack.org/python-neutronclient/queens/reference/index.html#

Here is the actual class/method implementation for floating_ips, can always look at the raw API documentation for neutron and not the python specific summary.

https://github.com/openstack/python-neutronclient/blob/master/neutronclient/neutron/v2_0/floatingip.py#L43

Here's where the novaclient bound methods are documented as DEPRECATED:
https://docs.openstack.org/python-novaclient/7.1.0/ref/v2/floating_ips.html

Reply:
Hi @mshriver, @lkhomenk is leaving us this Friday. I propose to merge this and work on neutron implementation in different PR. Do you agree?

Reply:
I agree, that would be great for you to contribute with neutron updates, but @psav  @jkrocil  @bsquizz @izapolsk and myself need to discuss merging #240 with the refactor.  Merging this will force @bsquizz to rebase that PR again.